subsys/bluetooth/mesh/adv.c

```diff
@@ -93,24 +103,42 @@ static struct net_buf *bt_mesh_adv_create_from_pool(struct net_buf_pool *buf_poo
 	adv->type         = type;
 	adv->tag          = tag;
 	adv->xmit         = xmit;
+	adv->hp           = hp;
 
 	return buf;
 }
 
-struct net_buf *bt_mesh_adv_create(enum bt_mesh_adv_type type,
-				   enum bt_mesh_adv_tag tag,
-				   uint8_t xmit, k_timeout_t timeout)
+struct net_buf *bt_mesh_adv_relay_create(bool hp, uint8_t xmit)
 {
 #if defined(CONFIG_BT_MESH_RELAY)
-	if (tag & BT_MESH_RELAY_ADV) {
-		return bt_mesh_adv_create_from_pool(&relay_buf_pool,
-						    adv_relay_pool, type,
-						    tag, xmit, timeout);
+#if defined(CONFIG_BT_MESH_RELAY_HP)
+	if (hp) {
+		struct net_buf *buf;
+
+		buf = bt_mesh_adv_create_from_pool(&relay_buf_hp_pool, adv_relay_hp_pool,
+						   BT_MESH_ADV_DATA, BT_MESH_RELAY_ADV,
+						   true, xmit, K_NO_WAIT);
+		if (buf) {
+			return buf;
+		}
```

Comment:
I only agree with some opinions. First of all, if the high-priority messages are too dense, resulting in insufficient buffers in the high-priority pool, then the message will be backed up as a normal relay message. This is the current practice, and what you mean is that even from  If the message is obtained from the ordinary relay pool, it can still be placed in the header of the relay_queue, so that the message can be relayed as quickly as possible. This is a good idea.

But I guess there is no worst case here, because for broadcast set it always takes priority to send messages from high priority.

Reply:
> The worst case is when the prioritized pool never gets empty and the messages from the non-prioritized pool are never sent.

I think this has exceeded the original intention of high priority. My expectation is to customize a limited buffer to send relays quickly, but if there are dense messages that require high-priority relays, then it is actually back to our problem.  early.